### ISSUE TITLE
form_tag doesn't compile as expected while using render :layout with erb partial

### ISSUE THREAD

**Issue by zernel** [indirect/haml-rails#118]
Hi there, I have got a problem when I try to render a erb partial as layout, actually there are 2 key elements to reproduce:
1. render erb partial as layout in haml template
2. form_tag in erb partial

Please check the sample code below:
```
# template.haml
= render layout: 'form' do
  %h4
    Test

# form.html.erb
<%= yield %>
<%= form_tag '/test' do -%>
  <%= text_field_tag :test %>
  <%= submit_tag %>
<%end -%>
```

this gets compiled to:
```
<h4>Test</h4>

<input type="text" name="test" id="test">
<input type="submit" name="commit" value="Save changes">
<form action="/test" accept-charset="UTF-8" method="post">
   <input name="utf8" type="hidden" value="✓"><input type="hidden" name="authenticity_token" value="...">
</form>
```

The form content were render in wrong place (outside of form tag), it will work correctly if I rewrite the `form.html.erb` to haml, I guess that maybe a bug.

**Comment by serv** [comment]
I am noticing a few things that's raising red flags.

- `<%= yield %>` is outside the `form_tag`. Maybe `input` tags are being inserted with `yield`.
- Change `<%end -%>` to just `<% end %>`

**Comment by zernel** [comment]
@serv good point, I try to move the `yield` into form_tag and even remove the `yield` from partial, neither is working (the `input` tags are still outside). And I try to change the `<% end %>` but still no lucky.
It will work if I write `form` tag manually instead using `form_tag` helper method. Something is wrong while rendering the `form_tag` in this case.

Anyway, thanks for your time :)

**Comment by zernel** [comment]
After digging more in the source code, I found the block became empty while it's execute in `form_tag`, but still not find the reason. quite weird.

**Comment by syguer** [comment]
@zernel please let me know version of rails and haml. Is it still happen on latest both version?
